Mengyuan Pull Request HW7

MengyuanZoe opened the pull request:
@hanghu @YangJiang89 @jhp312

MengyuanZoe committed:
Mengyuan Pull Request HW7

puleo-joe commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+
+import numpy as np
+import pandas as pd 
+import matplotlib.pyplot as plt
```
Both numpy and matplotlib are unused imports.  They can be deleted from the script.

puleo-joe commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+
+
+# Set global settings and macros.
+MAX_SHOW =10000
```
Add a space as follows: 
MAX_SHOW = 10000

puleo-joe commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+import os
```
Add a docstring to the beginning of the script

hanghu replied:
module name should all be in lowercase.
Quote from PEP 8:
"Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged."

puleo-joe commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+
+
+# Read in king county house data.
+data = pd.read_csv("../../Data/kc_house_data.csv", parse_dates = ['date'])
```
Remove whitespace in rename parse_date = ['date'] to parse_date=['date'] ...
Similar arguments should also remove whitespace

puleo-joe commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+data = pd.read_csv("../../Data/kc_house_data.csv", parse_dates = ['date'])
+data['zipcode'] = data['zipcode'].astype(str)
+
+## Create one map shwoing each listing of in house dataset and show in browser.
```
spelling correction:  change "shwoing" to "showing"

hanghu commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+
+# Save the house cluster map to a html and open it in broswer.
+houses_map.save(HOUSE_URL)
+webbrowser.open('file://' + os.path.realpath(HOUSE_URL),new=2)
```
there usually is one space after comma

hanghu commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+import matplotlib.pyplot as plt
+import folium 
+from folium import plugins
+import webbrowser
```
the standard import should be put in the first

hanghu commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+houses_heatmap.add_children(
+        plugins.HeatMap([[row["lat"], row["long"],row["price"]] 
+            for iters, row in data.iterrows()],
+            	min_opacity=0.5, max_zoom=18, radius = 8)) #[0:MAX_SHOW]
```
The about two lines use wrong continued indentation,

hanghu commented on HW7/Mengyuan_HW7.py:
```diff
@@ -0,0 +1,56 @@
+webbrowser.open('file://' + os.path.realpath(HOUSE_HEAT_URL),new=2)
+
+
+
```
The last couple of empty lines are redundant

MengyuanZoe committed:
Update Mengyuan_HW7.py to mengyuanhw7.py based on comments

All the comments are very helpful and good reminders of the coding rules. I have accepted them all and make changes accordingly.
